CORPORATION: adding prevState and changed state display in the warehouse UI

Caldwell-74 opened the pull request:
based on #857

Corp State now has getters to access nextState and prevState
prep work to include prevState in getCorporation() didnt include that yet 
because an API break is needed to change .state to .nextState to not cause confusion when .prevState is added

also changed how states are displayed in the warehouse ui to clear up the state system

![image](https://github.com/bitburner-official/bitburner-src/assets/115591472/1e5fc25d-c35b-48b3-84ac-ec3d1b125b6b)

Commit:
CORPORATION: fix division state display

remove this.state from division class
changing the calls to it to corporation.state.getState
to make sure the correct state variable is used
fixing that the ui shows the past state based on the out of date div.state while the api (getCorporation().state) shows the next state

Commit:
format,lint,,type

Commit:
groundwork for prev State and display

Commit:
progress bar length and load

Commit:
format

Commit:
getters

change state's states to getters

Commit:
CORPORATION: fix division state display

remove this.state from division class
changing the calls to it to corporation.state.getState
to make sure the correct state variable is used
fixing that the ui shows the past state based on the out of date div.state while the api (getCorporation().state) shows the next state

Commit:
format

Commit:
reformat

jjclark1982 commented on src/Corporation/CorporationState.ts:
```diff
@@ -9,20 +9,21 @@ export class CorporationState {
   // NOTE: This does NOT return the number stored in the 'state' property,
   // which is just an index for the array of all possible Corporation States.
-  getState(): CorpStateName {
+  get nextState(): CorpStateName {
```
maybe call these `state.nextName` and `state.prevName`?

Caldwell-74 replied:
what do you think about 
.nextStateName
.prevStateName ?

Caldwell-74 replied:
oh nvm the state is already in there yeah .nextName .prevName sound good

Caldwell-74 committed:
change getter Names and switch to util Function

Caldwell-74 commented:
there was the idea from myhf to use a different progress bar like
![image](https://github.com/bitburner-official/bitburner-src/assets/115591472/1d265d99-65a5-488d-8cbb-1c360f5a6ed4)
it would be a bit more complicated because the state name would need to move to "wrap" around correctly but if its wanted i can try to implement it

Caldwell-74 committed:
prettier,lint,doc

Commit:
deprecation of state

moved setDeprecationProperty to its own util file
added nextState and prevState to getCorporation
deprecated getCorporation().state

Commit:
better name

Snarling committed:
Merge branch 'dev' into prevState

Snarling commented:
Looks good! I just had to do a quick commit 0524112 to fix a conflict from #845

Caldwell-74 commented:
just for the future i run "npm run doc" by default
if i catch doc changes outside of the scope of my pr do you want me to make a new pr for those or just put them into that pr like with this?

Snarling commented:
If you already have doc changes in yours, just include all the fixes. If you don't have your own doc changes, any of the three here are acceptable:

1. Include the doc updates in your existing pr
2. Don't include them, make a separate pr
3. Don't include them, no separate pr

Ideally if any got missed, they'll be caught on someone's PR or on the next changelog update.
